src/wtc-gallery-component.js

```diff
@@ -325,46 +372,55 @@ class Gallery extends ElementController {
     }
 
     if (!next) {
-      console.warn('No item with index: ' + index);
+      console.warn("No item with index: " + index);
       return;
     }
 
     if (this.currentItem != next) {
-      this.currentItem.setAttribute('aria-hidden', 'true');
-      next.removeAttribute('aria-hidden');
-      _u.addClass('is-active is-transitioning is-transitioning--center', next);
-      _u.removeClass('is-active', this.currentItem);
+      this.currentItem.setAttribute("aria-hidden", "true");
+      next.removeAttribute("aria-hidden");
+      _u.addClass("is-active is-transitioning is-transitioning--center", next);
+      _u.removeClass("is-active", this.currentItem);
     }
 
     if (this.options.pagination) {
       _u.forEachNode(this.paginationItems, (counter, item) => {
         if (item.dataset.index == index) {
-          _u.addClass('is-active', item);
+          _u.addClass("is-active", item);
         } else {
-          _u.removeClass('is-active', item);
+          _u.removeClass("is-active", item);
         }
-      })
+      });
     }
 
-    if (typeof this.options.onHasChanged == "function") {
-      this.options.onHasChanged(next, this.currentItem);
+    if (this.liveRegion && this.options.liveRegionText) {
+      this.liveRegion.innerHTML = `${this.options.liveRegionText}: ${index +
+        1} of ${this.items.length}.`;
     }
 
-    this.liveRegion.innerHTML = `${this.options.liveRegionText}: ${index + 1} of ${this.items.length}.`;
+    const prev = this.currentItem;
     this.currentItem = next;
     this.currentIndex = index;
 
     if (!this.options.loop) {
       if (this.currentIndex == this.items.length - 1) {
-        this.nextBtn.setAttribute('disabled', true);
+        this.nextBtn.setAttribute("disabled", true);
       } else if (this.currentIndex == 0) {
-        this.prevBtn.setAttribute('disabled', true);
+        this.prevBtn.setAttribute("disabled", true);
       } else {
-        this.nextBtn.removeAttribute('disabled');
-        this.prevBtn.removeAttribute('disabled');
+        this.nextBtn.removeAttribute("disabled");
+        this.prevBtn.removeAttribute("disabled");
       }
     }
 
+    if (typeof this.options.onHasChanged == "function") {
+      this.options.onHasChanged({
```

Comment:
While I love this approach, let's make it three separate parameters (in the same order as before, plus the new one), rather than one object. This will avoid any breaking changes on old projects that use `hasChanged`.